plugins/_transforms.js

```diff
@@ -164,114 +164,104 @@ const mth = {
 
 /**
  * Decompose matrix into simple transforms.
- *
  * @param {TransformItem} transform
- * @param {TransformParams} params
  * @returns {TransformItem[]}
  * @see https://frederic-wang.fr/decomposition-of-2d-transform-matrices.html
  */
-export const matrixToTransform = (transform, params) => {
-  const floatPrecision = params.floatPrecision;
+export const matrixToTransform = (transform) => {
   const data = transform.data;
+
+  // Where applicable, variables are named in accordance with the frederic-wang document referenced above.
+  const a = data[0];
+  const b = data[1];
+  const c = data[2];
+  const d = data[3];
+  const e = data[4];
+  const f = data[5];
+  const delta = a * d - b * c;
+  if (delta === 0) {
+    return [transform];
+  }
+  const r = Math.hypot(a, b);
+  if (r === 0) {
+    return [transform];
+  }
+  const angle_cos = a / r;
```

Comment:
Nitpicking, but could you use camelcase variables in your patch for consistency?

Reply:
Will do. Would be nice if the linter could catch this.